src/main/java/sparksoniq/jsoniq/tuple/FlworKey.java

```diff
@@ -90,13 +90,6 @@ public int compareWithFlworKey(FlworKey flworKey) {
             ) {
                 throw new SparksoniqRuntimeException("Non atomic key not allowed");
             }
```

Comment:
Is there a reason why this check is being removed -- is this checked somewhere else (e.g., the compareTo() call)?

Reply:
Thank you for noticing. Indeed we cannot use such check anymore as we want to allow the comparison between items with different class names (e.g. DayTimeDuration and Duration). Even though, as you rightfully mentioned, it is compareTo() function that takes care of the current comparison fashion, I still need to add a catch for exception, in case this is thrown by compareTo(). I added such check for dataframes as well. 
Thanks you